presto-main/src/main/java/io/prestosql/cost/CostCalculatorUsingExchanges.java

```diff
@@ -234,67 +247,127 @@ public PlanNodeLocalCostEstimate visitExchange(ExchangeNode node, Void context)
+                    .reduce(PlanCostEstimate.zero(), CostCalculatorUsingExchanges::addSiblingsCost);
+            return new PlanCostEstimate(
+                    sourcesCost.getCpuCost() + localCost.getCpuCost(),
+                    max(
+                            sourcesCost.getMaxMemory(),
+                            sourcesCost.getMaxMemoryWhenOutputting() + localCost.getMaxMemory()),
+                    localCost.getMaxMemory(),
+                    sourcesCost.getNetworkCost() + localCost.getNetworkCost());
+        }
+
+        private PlanCostEstimate costForStreaming(PlanNode node, PlanNodeLocalCostEstimate localCost)
+        {
+            PlanCostEstimate sourcesCost = getSourcesEstimations(node).stream()
+                    .reduce(PlanCostEstimate.zero(), CostCalculatorUsingExchanges::addSiblingsCost);
+            return new PlanCostEstimate(
+                    sourcesCost.getCpuCost() + localCost.getCpuCost(),
+                    max(
+                            sourcesCost.getMaxMemory(),
+                            sourcesCost.getMaxMemoryWhenOutputting() + localCost.getMaxMemory()),
+                    sourcesCost.getMaxMemoryWhenOutputting() + localCost.getMaxMemory(),
+                    sourcesCost.getNetworkCost() + localCost.getNetworkCost());
+        }
+
+        private PlanCostEstimate costForLookupJoin(PlanNode node, PlanNodeLocalCostEstimate localCost)
+        {
+            verify(node.getSources().size() == 2, "Unexpected number of sources for %s: %s", node, node.getSources());
+            List<PlanCostEstimate> sourcesCosts = getSourcesEstimations(node);
+            verify(sourcesCosts.size() == 2);
+            PlanCostEstimate probeCost = sourcesCosts.get(0);
+            PlanCostEstimate buildCost = sourcesCosts.get(1);
+
+            return new PlanCostEstimate(
+                    probeCost.getCpuCost() + buildCost.getCpuCost() + localCost.getCpuCost(),
+                    max(
+                            probeCost.getMaxMemory() + buildCost.getMaxMemory(),
+                            probeCost.getMaxMemoryWhenOutputting() + buildCost.getMaxMemoryWhenOutputting() + localCost.getMaxMemory()),
+                    probeCost.getMaxMemoryWhenOutputting() + localCost.getMaxMemory(),
+                    probeCost.getNetworkCost() + buildCost.getNetworkCost() + localCost.getNetworkCost());
         }
 
         private PlanNodeStatsEstimate getStats(PlanNode node)
         {
             return stats.getStats(node);
         }
+
+        private List<PlanCostEstimate> getSourcesEstimations(PlanNode node)
+        {
+            return node.getSources().stream()
+                    .map(sourcesCosts::getCost)
+                    .collect(toImmutableList());
+        }
     }
 
-    private static PlanCostEstimate add(PlanCostEstimate a, PlanCostEstimate b)
+    private static PlanCostEstimate addSiblingsCost(PlanCostEstimate a, PlanCostEstimate b)
```

Comment:
this is awfully similar to: https://github.com/prestosql/presto/pull/247/commits/339eb652dfae13e98d07de7bae8383c496db9536#diff-4a41ea165b82f6cedc978ea51f51fbf1R82

Reply:
that method no longer exists